realm/realm-library/src/androidTest/java/io/realm/ObjectChangeSetTests.java

```diff
@@ -417,4 +418,49 @@ public void onChange(DynamicRealmObject object, ObjectChangeSet changeSet) {
         }
         realm.commitTransaction();
     }
+
+    // For https://github.com/realm/realm-java/issues/4474
+    @Test
+    @RunTestInLooperThread
+    public void allParentObjectShouldBeInChangeSet() {
+        Realm realm = looperThread.realm;
+
+        realm.beginTransaction();
+        Owner owner = realm.createObject(Owner.class);
+        Dog dog1 = realm.createObject(Dog.class);
+        dog1.setHasTail(true);
```

Comment:
Swap this line with the next so `setHasTail/setOwner` comes in the same order for all dogs